### PR TITLE
Reorganize R4 Resources, Add Overview Links

### PR DESCRIPTION
**Pull request by beichhor**
Description
----
The changes in this PR are to accomplish two primary goals:
a) Organize the fhir.cerner.com resources for R4 to make sense and match how they are organized via the FHIR spec
b) Have the Overview table contain links automatically to the resources. The existing resource list in this PR was automatically generated from scraping the HL7 Resource list site https://www.hl7.org/fhir/r4/resourcelist.html The "Basic" resource link was removed (because it currently only links to FinancialTransaction but this helps avoid confusion) and the FinancialTransaction resource was added manually.

The included screenshots are of the primary location of the changes (the overview table and sidebar to capture the organization) but due to the reorganization effort it won't be possible to capture a complete set of screenshots.

Screen Shot 2021-04-14 at 3 50 00 PM
Screen Shot 2021-04-14 at 3 49 40 PM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: content/millennium/r4/base/workflow.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: Workflow | R4 API
+layout: api
+---
+
+# Individuals
```
Comment (Coffeevee) [inline content/millennium/r4/base/workflow.md:6]: Should probably be `# Workflow`

File: content/millennium/r4/base/entities.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+title: Entities | R4 API
+layout: api
+---
+
+# Entities
+
+<%= render '/millennium/r4/base/entities.*' %>
+
+* [Organization](../base/entities/organization)
+* [Location](../base/entities/location)
```
Comment (Coffeevee) [inline content/millennium/r4/base/entities.md:11]: I think we're missing the `[Device]` item on this list

File: content/millennium/r4/foundation/conformance.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+title: Conformance | R4 API
+layout: api
+---
+
+# Conformance
+
+<%= render '/millennium/r4/foundation/conformance.*' %>
+
+* [Capability Statement](../foundation/conformance/capability-statement)
+* [Structure Definition](../foundation/conformance/structure-definition)
```
Comment (Coffeevee) [inline content/millennium/r4/foundation/conformance.md:11]: I think we're missing `[Operation Definition]` here

Comment (beichhor) [reply content/millennium/r4/foundation/conformance.md:11]: https://github.com/cerner/fhir.cerner.com/commit/775cdd410a50bbc809a19a30052924b93f1fd20f

File: content/millennium/r4/base/entities.md (hunk 2 above)
Comment (beichhor) [reply content/millennium/r4/base/entities.md:11]: https://github.com/cerner/fhir.cerner.com/commit/775cdd410a50bbc809a19a30052924b93f1fd20f

File: content/millennium/r4/base/workflow.md (hunk 1 above)
Comment (beichhor) [reply content/millennium/r4/base/workflow.md]: https://github.com/cerner/fhir.cerner.com/commit/775cdd410a50bbc809a19a30052924b93f1fd20f

File: content/millennium/r4/clinical.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+<%= render '/millennium/r4/clinical/care-provision.*' %>
+
+* [CarePlan](../clinical/care-provision/care-plan)
+* [CareTeam](../clinical/care-provision/care-team)
+* [Goal](../clinical/care-provision/goal)
+* [ServiceRequest](../clinical/care-provision/service-request)
+* [NutritionOrder](../clinical/care-provision/nutrition-order)
+
+
```
Comment (Coffeevee) [inline content/millennium/r4/clinical.md:38]: nit - remove extra line

**Comment by Coffeevee** [comment]
Because this is a core part of site navigation, I think it would improve readability to alphabetize the items in each of the resource lists.

For example, currently within Foundation/Conformance, the listed items are:
> CapabilityStatement
> StructureDefinition
> OperationDefinition

We would update this list to be:
> CapabilityStatement
> OperationDefinition
> StructureDefinition

Thoughts?

File: nanoc.yaml [hunk 5]
```diff
@@ -193,6 +193,106 @@ redirects:
...
+  /millennium/r4/encounters:
+    url: /millennium/r4/base/management
+  /millennium/r4/encounters/encounter:
+    url: /millennium/r4/base/management/encounter
+  /millennium/r4/entities:
+    url: /millennium/r4/base/entities
+  /millennium/r4/entities/location:
+    url: /millennium/r4/base/entities/location
+  /millennium/r4/entities/organization:
+    url: /millennium/r4/base/entities/organization
```
Comment (Coffeevee) [inline nanoc.yaml:244]: I think we're missing the `entities/device` item here

**Comment by beichhor** [comment]
> Because this is a core part of site navigation, I think it would improve readability to alphabetize the items in each of the resource lists.
> 
> For example, currently within Foundation/Conformance, the listed items are:
> 
> > CapabilityStatement
> > StructureDefinition
> > OperationDefinition
> 
> We would update this list to be:
> 
> > CapabilityStatement
> > OperationDefinition
> > StructureDefinition
> 
> Thoughts?

I'm mostly against this because that will deviate from how the [FHIR spec itself](https://www.hl7.org/fhir/r4/resourcelist.html) organizes the resources, which from a usability standpoint makes sense (ie, having CapabilityStatement listed first). I intentionally am trying to match the ordering they display.

**Comment by Coffeevee** [comment]
> > Because this is a core part of site navigation, I think it would improve readability to alphabetize the items in each of the resource lists.
> > For example, currently within Foundation/Conformance, the listed items are:
> > > CapabilityStatement
> > > StructureDefinition
> > > OperationDefinition
> > 
> > 
> > We would update this list to be:
> > > CapabilityStatement
> > > OperationDefinition
> > > StructureDefinition
> > 
> > 
> > Thoughts?
> 
> I'm mostly against this because that will deviate from how the [FHIR spec itself](https://www.hl7.org/fhir/r4/resourcelist.html) organizes the resources, which from a usability standpoint makes sense (ie, having CapabilityStatement listed first). I intentionally am trying to match the ordering they display.

Makes sense. Thanks for explaining your reasoning.

**Review by amcginn — APPROVED** [review]

**Review by Coffeevee — APPROVED** [review]

**Comment by amcginn** [comment]
as we rebase, we need to ensure that any resources that have been added since these original changes are added in the appropriate locations

**Comment by amcginn** [comment]
for reference, these new categories seem to be following what HL7 has listed here: https://hl7.org/fhir/R4/resourcelist.html

**Comment by amcginn** [comment]
resources are being reorganized as part of the OHC migration, so this will not be done here.

**Pull request closed without merging**